### PR TITLE
Introduce downtime#scheduled_by

### LINKED ISSUE #231: Downtime tables should include downtime source/origin
Just faced again the issue that [Icinga 2 does not complain about a failed downtime removal](https://github.com/Icinga/icingaweb2/issues/4247) if it's coming from a scheduled downtime config object.

Icinga DB Web could hide the removal button in such a case, though there doesn't seem to be a column in the schema indicating what created the downtime. I'd like to have that included.

### PR DESCRIPTION
**Pull request by Al2Klimov**
fixes #231
closes #257

### CODE REVIEW

**Comment by Al2Klimov** [comment]
## Blocked by

* [ ] Icinga/icinga2#8631

File: schema/mysql/schema.sql [hunk 1]
```diff
@@ -588,6 +588,7 @@ CREATE TABLE downtime (
   is_in_effect enum('n', 'y') NOT NULL,
   start_time bigint unsigned DEFAULT NULL COMMENT 'Time when the host went into a problem state during the downtimes timeframe',
   end_time bigint unsigned DEFAULT NULL COMMENT 'Problem state assumed: scheduled_end_time if fixed, start_time + flexible_duration otherwise',
+  scheduled_by varchar(255) DEFAULT NULL,
```
Comment (julianbrost) [inline schema/mysql/schema.sql:591]: 255 is probably not long enough. The value will be "host_name!service_name!downtime_name", so probably something like 3*255 + 2 is required.

**Review by julianbrost — CHANGES_REQUESTED** [review]

File: schema/mysql/schema.sql (hunk 1 above)
Comment (lippserd) [reply schema/mysql/schema.sql:591]: And there should be a `COMMENT` that explains that and what the column reflects.

Comment (Al2Klimov) [reply schema/mysql/schema.sql:591]: @lippserd Can you confirm that length?

Comment (lippserd) [reply schema/mysql/schema.sql:591]: We have no limit on host and service names other than the implicit file name length limit for API-created objects, right?
I think this should be a text column just to be on the safe side. MySQL will not execute the statement if values need to be truncated and it is running in strict mode. Which in turn would stop Icinga DB.

Comment (julianbrost) [reply schema/mysql/schema.sql:591]: `host.name` and `service.name` are both `varchar(255)`. Assuming the same length for the names of ScheduledDowntimes, that yields the limit I suggested above. Haven't checked if Icinga 2 enforces these limits, but if it doesn't, maybe it should if we have database schemas with those limits?

Comment (lippserd) [reply schema/mysql/schema.sql:591]: If the downtime name can't be longer than 255 chars 3*255+2 is fine. For host and service names we already have the 255 char length restriction in other tables. But yes, Icinga 2 should enforce those limits.

Comment (Al2Klimov) [reply schema/mysql/schema.sql:591]: I.e. Icinga 2 shall reject config of objects w/ names >255?

**Comment by Al2Klimov** [comment]
## TODO

* [ ] Reason column length

**Review by lippserd — CHANGES_REQUESTED** [review]
I would rewrite the comment to

> Name of the ScheduledDowntime which created this downtime. 255+1+255+1+255, i.e. "host.name!service.name!scheduled-downtime-name".

**Review by lippserd — APPROVED** [review]

**Review by lippserd — CHANGES_REQUESTED** [review]
Please also add it to the history.

**Comment by Al2Klimov** [comment]
## TODO

* https://github.com/Icinga/icinga2/pull/8966

**Comment by Al2Klimov** [comment]
## Purrs

```
MariaDB [idb]> select scheduled_by from downtime_history;
+---------------------------------------------------+
| scheduled_by                                      |
+---------------------------------------------------+
| Alexanders-MacBook-Pro.local!load!backup-downtime |
+---------------------------------------------------+
1 row in set (0.004 sec)

MariaDB [idb]>
```

**Review by julianbrost — APPROVED** [review]

**Review by bobapple — APPROVED** [review]

**Review by lippserd — APPROVED** [review]

**Pull request merged**